apps/studio/pages/project/[ref]/functions/index.tsx

```diff
@@ -38,6 +54,33 @@ const EdgeFunctionsPage: NextPageWithLayout = () => {
     isSuccess,
   } = useEdgeFunctionsQuery({ projectRef: ref })
 
+  const [searchString, setSearchString] = useState('')
+  const [sort, setSort] = useState<EdgeFunctionsSort>('name:asc')
+
+  const filteredFunctions = useMemo(() => {
+    const temp = (functions ?? []).filter((x) =>
+      x.name.toLowerCase().includes(searchString.toLowerCase())
+    )
+    const [sortCol, sortOrder] = sort.split(':') as [
+      EdgeFunctionsSortColumn,
+      EdgeFunctionsSortOrder,
+    ]
+    const orderMultiplier = sortOrder === 'asc' ? 1 : -1
+
+    return temp.sort((a, b) => {
+      if (sortCol === 'name') {
+        return a.name.localeCompare(b.name) * orderMultiplier
+      }
+      if (sortCol === 'created_at') {
+        return (a.created_at - b.created_at) * orderMultiplier
+      }
+      if (sortCol === 'updated_at') {
+        return (a.updated_at - b.updated_at) * orderMultiplier
+      }
+      return 0
+    })
```

Comment:
this is a bit better when using strings, makes the types harder to mess up and make the implementation easier